src/util.jl

```diff
@@ -318,3 +318,6 @@ function tuplesetindex(x::Union{NamedTuple, Tuple}, v::Tuple, i::Tuple)
     reduce((t, j)->tuplesetindex(t, v[j], i[j]), x, 1:length(i))
 end
 
+
+# mild piracy
+Base.similar(::PooledArrays.PooledArray, ::Type{DataValue{T}}, sz::Tuple{Vararg{Int64,N}}) where {T,N} = DataValueArray{T}(sz)
```

Comment:
Hehe. I'm starting to wonder whether the method I'm adding to ``similar`` in DataValues.jl is problematic... But if this fixes it for now, I'd call it a day.